samples/emitc_modules/BUILD

```diff
@@ -0,0 +1,83 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+load("//build_tools/bazel:build_defs.oss.bzl", "iree_cmake_extra_content", "iree_runtime_cc_library", "iree_runtime_cc_test")
+load("//build_tools/bazel:iree_c_module.bzl", "iree_c_module")
+
+package(
+    default_visibility = ["//visibility:public"],
+    features = ["layering_check"],
+    licenses = ["notice"],  # Apache 2.0
+)
+
+iree_cmake_extra_content(
+    content = """
+if(NOT IREE_ENABLE_EMITC)
+  return()
+endif()
+""",
+    inline = True,
+)
+
+iree_c_module(
+  name = "add_module",
+  src = "add.mlir",
+  h_file_output = "add_module.h",
+  flags = [
+    "--iree-mlir-to-vm-c-module",
+    "--iree-hal-target-backends=vmvx"
+  ],
+)
```

Comment:
This usage of flags is inconsistent.

https://github.com/google/iree/blob/931b405fee3a6bd76da969ed2ad4313602ee9301/compiler/src/iree/compiler/Tools/iree_compile_lib.cc#L181-L189

```
def iree_c_module(
        ...
        flags = [
          "--compile-mode=vm",
          "--output-format=vm-c",
        ],
        ...
        flags: additional flags to pass to the compiler. Bytecode
            translation and backend flags are passed automatically.
```

The new (non-deprecated) flags should be `--compile-mode=vm --output-format=vm-c`. I think in this case, the ` --iree-hal-target-backends=vmvx` flag isn't even useful (the samples don't use the HAL). The `iree_c_module` should always pass at least `--output-format=vm-c` and `iree_bytecode_module` should always pass at least `--output-format=vm-bytecode`. The compile mode (from VM IR or from standard input dialects) can be customized, as can the HAL target backends.

So I think for this PR, I'd just match https://github.com/google/iree/blob/main/build_tools/cmake/iree_c_module.cmake (no default flags), but there's more cleanup to do here (and across the project).